Flutter_TDD_Architecture_Course_notes.md

```diff
@@ -1013,3 +1013,402 @@ class NetworkInfoImpl implements NetworkInfo {
   Future<bool> get isConnected => connectionChecker.hasConnection;
 }
 ```
+#### 8.  Local Data Source
+> Video 8: https://www.youtube.com/watch?v=fCguzcvLka8
+
+In the `eighth` tutorial the shared_preferences function is implemented.
+
+It uses the `shared_preferences` function because we only need to save one given NumberTrivia.
+
+Let's add the code to the `number_trivia_local_data_source_test.dart` file.
+
+```dart
+import 'package:flutter_test/flutter_test.dart';
+import 'package:mockito/mockito.dart';
+import 'package:matcher/matcher.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+import '../../../../fixtures/fixture_reader.dart';
+
+class MockSharedPreferences extends Mock implements SharedPreferences {}
+
+void main() {
+  NumberTriviaLocalDataSourceImpl dataSource;
+  MockSharedPreferences mockSharedPreferences;
+
+  setUp(() {
+    mockSharedPreferences = MockSharedPreferences();
+    dataSource = NumberTriviaLocalDataSourceImpl(
+      sharedPreferences: mockSharedPreferences,
+    );
+  });
+}
+```
+Now the file corresponding to the `number_trivia_local_data_source.dart`.
+
+```dart
+...
+class NumberTriviaLocalDataSourceImpl implements NumberTriviaLocalDataSource {
+  final SharedPreferences sharedPreferences;
+
+  NumberTriviaLocalDataSourceImpl({@required this.sharedPreferences});
+
+  @override
+  Future<NumberTriviaModel> getLastNumberTrivia() {
+    // TODO: implement getLastNumberTrivia
+    return null;
+  }
+
+  @override
+  Future<void> cacheNumberTrivia(NumberTriviaModel triviaToCache) {
+    // TODO: implement cacheNumberTrivia
+    return null;
+  }
+}
+```
+Then we created another `fixture` with the name `trivia_cached.json`. With the code.
+
+```dart
+{
+  "text": "Test Text",
+  "number": 1
+}
+```
+#### getLastNumberTrivia
+
+We now make the first test, with the first feature that will show a number stored in the cache.
```

Comment:
This sentence is ordered like it was written in PT and then translated to EN.  It can be modified to:
```suggestion
Create the first test for the first feature that will show a number stored in the cache.
```